src/thing_shots.c

```diff
@@ -472,9 +476,42 @@ TbBool shot_hit_wall_at(struct Thing *shotng, struct Coord3d *pos)
                 apply_damage_to_thing(doortng, i, shotst->damage_type, -1);
             } else
             {
-                efftng = create_shot_hit_effect(&shotng->mappos, shotng->owner, shotst->hit_generic.effect_model, shotst->hit_generic.sndsample_idx, shotst->hit_generic.sndsample_range);
-                if (!shotst->hit_generic.withstand)
+                long smpl_idx;
+                long range;
+                long eff_kind;
+                struct Coord3d pos;
```

Comment:
This line throws a build warning, because `pos` already exists. You get it with the function.